src/Microsoft.TemplateEngine.Edge/Settings/ComponentManager.cs

```diff
@@ -55,48 +44,15 @@ public ComponentManager(ISettingsLoader loader, SettingsStore userSettings)
 
             foreach (KeyValuePair<string, string> entry in userSettings.ComponentGuidToAssemblyQualifiedName)
             {
-                if (Guid.TryParse(entry.Key, out Guid componentId) && allowedIds.Contains(componentId))
+                if (Guid.TryParse(entry.Key, out Guid componentId))
                 {
                     _componentIdToAssemblyQualifiedTypeName[componentId] = entry.Value;
                 }
             }
 
-            if (!allowedIds.Contains(FileSystemMountPointFactory.FactoryId))
-            {
-                allowedIds.Add(FileSystemMountPointFactory.FactoryId);
-                AddComponent(typeof(IMountPointFactory), new FileSystemMountPointFactory());
-            }
-
-            if (!allowedIds.Contains(ZipFileMountPointFactory.FactoryId))
-            {
-                allowedIds.Add(ZipFileMountPointFactory.FactoryId);
-                AddComponent(typeof(IMountPointFactory), new ZipFileMountPointFactory());
-            }
-
-            if (!allowedIds.Contains(GlobalSettingsTemplatePackageProviderFactory.FactoryId))
-            {
-                allowedIds.Add(GlobalSettingsTemplatePackageProviderFactory.FactoryId);
-                AddComponent(typeof(ITemplatePackageProviderFactory), new GlobalSettingsTemplatePackageProviderFactory());
-            }
-
-            if (!allowedIds.Contains(NuGetInstallerFactory.FactoryId))
+            foreach (var (interfaceType, instance) in _loader.EnvironmentSettings.Host.BuiltInComponents)
```

Comment:
should we dedup here?

Reply:
Dedup already happens inside "AddComponent" method when it is adding to `ComponentCache`. Problem with this de-dup logic was that if same Instance implemented two interfaces, it now failed...